Services/AspireService.cs

```diff
@@ -1,93 +1,163 @@
+                }
+
+                var result = labels.TryGetValue("containerapps.io/app-name", out string appName);
+                if (!result)
+                {
+                    _logger.LogDebug($"The Pod {item.Metadata.Name} does not have an entry for the key app. Skipping to next pod.");
+                    return;
+                }
+
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                if (type.Equals(WatchEventType.Added) || type.Equals(WatchEventType.Modified))
+                {
+                    watchResourcesUpdate.Changes.Value.Add(new WatchResourcesChange() { Upsert = { Name = appName, DisplayName = appName, ResourceType = item.Kind, Uid = item.Uid(), CreatedAt = Google.Protobuf.WellKnownTypes.Timestamp.FromDateTime(item.CreationTimestamp().Value), State = item.Status.Phase, Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = item.Status.PodIP } }, Properties = { new ResourceProperty() { DisplayName = item.Kind, Name = item.Metadata.Name } } } });
+                    await responseStream.WriteAsync(watchResourcesUpdate);
+                }
+                else if (type.Equals(WatchEventType.Deleted))
+                {
+                    watchResourcesUpdate.Changes.Value.Add(new WatchResourcesChange() { Delete = new ResourceDeletion { ResourceName = item.Metadata.Name, ResourceType = item.Kind } });
+                    await responseStream.WriteAsync(watchResourcesUpdate);
+                }
+            });
+        }
+
+        public override async Task WatchResourceConsoleLogs(WatchResourceConsoleLogsRequest request, IServerStreamWriter<WatchResourceConsoleLogsUpdate> responseStream, ServerCallContext context)
+        {
+            var label = $"containerapps.io/app-name={request.ResourceName}";
+            _logger.LogInformation($"Label : {label}");
+            var pods = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace, labelSelector: label);
+
+            // We only have one pod deployed for each Aspire App
+            if (pods.Items.Count() > 1)
+            {
+                _logger.LogDebug($"Multiple pods were found for the app {request.ResourceName}");
+            }
+
+            var podName = pods.Items[0].Metadata.Name;
+            var container = pods.Items[0].Spec.Containers.Where(c => c.Name == request.ResourceName).FirstOrDefault();
+            if (container == null)
+            {
+                _logger.LogWarning($"Container matching {request.ResourceName} does not exist");
+                return;
+            }
+            
+            _logger.LogInformation($"PodName: {podName}");
+            var stream = new System.IO.MemoryStream();
+            while (!context.CancellationToken.IsCancellationRequested)
+            {
+                // Re-set stream to the beginning to make sure repetitive logs are not read. 
+                stream.Seek(stream.Length, SeekOrigin.Begin);

```

Comment:
Is the comment valid with the code? The comment says that we're putting back the stream to the beginning, but I believe that the code is putting it at `stream.Length`, so the end?

Reply:
Updated this line so the stream is reset to the beginning. The intention here is to re-set the stream before we read into it in the next line.

Reply:
Sorry I hope I'm not confused, but then aren't we going to read all over again from the beginning?

So basically if we have these logs in a pod:
LogA
LogB
LogC

We call this method once, read the three logs and reach EndOfStream.

Then another log, LogD, gets added. Wouldn't we return all over again all the logs from the start?
LogA
LogB
LogC
LogD